docs/rules/generator-star-spacing.md

```diff
@@ -75,6 +75,25 @@ An example of shorthand configuration:
 "generator-star-spacing": ["error", "after"]
 ```
 
+Additionally, this rule allows further configurability via overrides per function type.
+
+* `named` provides overrides for named functions
+* `anonymous` provides overrides for anonymous functions
+* `method` provides overrides for class methods or property function shorthand
+* `static` provides overrides for static class methods
+
+An example of a configuration with overrides:
+
+```json
+"generator-star-spacing": ["error", {
+    "before": false,
+    "after": true,
+    "anonymous": "neither",
+    "method": "neither",
+    "static": {"before": true, "after": true}
+}]
```

Comment:
Can you explain what this configuration does? In particular, I'm unsure about what happens when `"before"` and `"after"` are used as top-level options alongside the more configurable options.